### PR TITLE
Database from schema

### PR DESCRIPTION
**Pull request by SFarida**
### This PR contains sql commands to create the following tables based on a schema diagram
- patients
- treatments
- medical_histories
- invoices
- invoice_items
image

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+   CONSTRAINT fk_patient
+      FOREIGN KEY(patient_id) 
+	  REFERENCES patients(id)
+	  ON DELETE CASCADE
+);
+
+CREATE TABLE invoices(
+   id serial PRIMARY KEY,
+   medical_history_id INT,
+   total_amount REAL,
```
Comment (christianbiring1) [inline schema_based_on_diagram.sql]: [ OPTIONAL]
- [ ] The `decimal` type would be better here as specified in the table
```suggestion
   total_amount DECIMAL,
```

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+   date_of_birth DATE
+);
+
+CREATE TABLE treatments(
+   id serial PRIMARY KEY,
+   type VARCHAR(100),
+   name VARCHAR(250)
+);
+
+CREATE TABLE medical_histories(
```
Comment (christianbiring1) [inline schema_based_on_diagram.sql:13]: - [ ] Kindly make sure the `many-to-many` relationship between the medical_history and treatments is creating through an intermediate table.
Hint: You may consider saying:
```sql
CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```

Comment (christianbiring1) [reply schema_based_on_diagram.sql:13]: - [ ] Also make sure all your foreign keys have their indexes as specified in the project requirements.
e.g:
```sql
CREATE INDEX ON medical_histories (patient_id);
-- and so on --
```

**Review by christianbiring1 — CHANGES_REQUESTED** [review]
Hi @SFarida @AyomipoMajek , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
# REQUEST CHANGES :recycle: 
soclose

## Highlights
- [ ] Descriptive PR `title` and `summary` :+1: 
- [ ] All tables created correctly :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (SFarida) [reply schema_based_on_diagram.sql]: Hello @christianbiring1 thank you for the remarks, I will correct and commit the changes.

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+CREATE TABLE medical_histories_has_treatments (
+    medical_history_id INT REFERENCES medical_histories(id),
+    treatment_id INT REFERENCES treatments(id)
+);
+
+CREATE INDEX ON medical_histories (patient_id);
+CREATE INDEX ON invoices (medical_history_id);
+CREATE INDEX ON invoice_items (invoice_id);
+CREATE INDEX ON invoice_items (treatment_id);
```
Comment (Ismailco) [inline schema_based_on_diagram.sql]: - [ ] Greate work making the suggested changes by the previous reviewer however you should create indexes for the new table between `medical_histories` and `treatments`, `medical_histories_has_treatments`. kindly consider adding INDEX for `medical_histories_has_treatments (medical_history_id);` and `medical_histories_has_treatments (treatment_id);`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @SFarida, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights

✔️ Implement changes
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 3 above)
Comment (SFarida) [reply schema_based_on_diagram.sql]: Thank you @Ismailco I was not keen enough I corrected it.

**Review by Munsa1 — APPROVED** [review]
Hi @SFarida, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Required Changes :heavy_check_mark: 
Good Communication With Reviewers :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**